### PR TITLE
Improvements/DI module

### PR DESCRIPTION
**Pull request by astubbs**
Dependency injection, needed for sane testing of transactions. 

Abiding by the zero dependencies policy - just needed something simple.

- [ ] Changelog

### CODE REVIEW

**Review by astubbs — COMMENTED** [review]
LGTM

**Review by rkolesnev — APPROVED** [review]
LGTM - couple of questions - added as comments

File: parallel-consumer-core/src/main/java/io/confluent/parallelconsumer/internal/PCModule.java [hunk 1]
```diff
@@ -0,0 +1,94 @@
+package io.confluent.parallelconsumer.internal;
```
[the remark this answers is not in the document]
Comment (rkolesnev) [reply parallel-consumer-core/src/main/java/io/confluent/parallelconsumer/internal/PCModule.java:1]: nvm - overrides can be provided by extending PCModule class and passing that into EOS Processor constructor on creation.

File: parallel-consumer-core/src/main/java/io/confluent/parallelconsumer/internal/PCModule.java [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+    protected ProducerManager<K, V> producerManager() {
+        if (producerManager == null) {
+            this.producerManager = new ProducerManager<>(producer(), consumerManager(), workManager(), options());
+        }
+        return producerManager;
+    }
+
+    public Producer<K, V> producer() {
+        return optionsInstance.getProducer();
```
[the remark this answers is not in the document]
Comment (astubbs) [reply parallel-consumer-core/src/main/java/io/confluent/parallelconsumer/internal/PCModule.java:47]: discussed

File: parallel-consumer-core/src/main/java/io/confluent/parallelconsumer/internal/PCModule.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (astubbs) [reply parallel-consumer-core/src/main/java/io/confluent/parallelconsumer/internal/PCModule.java:1]: discussed

**Pull request merged**